src/server/frontend_wayland/wl_client.cpp

```diff
@@ -112,25 +114,58 @@ void mf::WlClient::setup_new_client_handler(
     wl_display_add_destroy_listener(display, &context->display_destruction_listener);
 }
 
-auto mf::WlClient::from(wl_client* client) -> WlClient*
+auto mf::WlClient::from(wl_client* client) -> WlClient&
 {
     if (!client)
     {
-        return nullptr;
+        fatal_error("WlClient::from(): wl_client is null");
     }
     auto listener = wl_client_get_destroy_listener(client, &cleanup_client_ctx);
+    if (!listener)
+    {
+        fatal_error("WlClient::from(): listener is null");
+    }
     auto ctx = ClientCtx::from(listener);
-    return ctx ? ctx->client.get() : nullptr;
+    if (!ctx)
+    {
+        fatal_error("WlClient::from(): ctx is null");
+    }
+    return *ctx->client.get();
```

Comment:
How sure are we that `client.get()` will never be `nullptr`?

Reply:
I believe fairly sure